src/vs/platform/sharedProcess/electron-main/sharedProcess.ts

```diff
@@ -241,6 +243,8 @@ export class SharedProcess extends Disposable implements ISharedProcess {
 			appRoot: this.environmentMainService.appRoot,
 			codeCachePath: this.environmentMainService.codeCachePath,
 			backupWorkspacesPath: this.environmentMainService.backupWorkspacesPath,
+			defaultProfile: this.userDataProfilesService.defaultProfile,
```

Comment:
Please double check that `URI` is properly transfered. All the other types seem to be primitive types, and now you add `URI`.

Reply:
Good point and yes I am handling this already - https://github.com/microsoft/vscode/blob/5178151d231622a5ec607890f49e3d9d47bcb377/src/vs/code/electron-browser/sharedProcess/sharedProcessMain.ts#L234

Reply:
Ideally this is not something users have to do when working with the config, it should work transparently via the provider of configuration. 

In other words: if the type is declared as `URI`, I would not expect that as a consumer I have to call `revive`.